mypy/checker.py

```diff
@@ -4378,6 +4380,75 @@ def infer_issubclass_maps(self, node: CallExpr,
         yes_map, no_map = map(convert_to_typetype, (yes_map, no_map))
         return yes_map, no_map
 
+    def check_long_tuple_assignment(self,
```

Comment:
Since this mostly about generating error messages, I think that it would be better to move this method (and the methods below) to `mypy.messages.MessageBuilder`.

Also, the method name can be a bit misleading since elsewhere 'check' means 'type check', whereas here it seems to mean something else. Maybe use a `try_` prefix instead. Random idea: `try_report_long_tuple_assignment_error`.